jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -100,9 +101,15 @@ public CaseDefinitionSummary getCaseDefinition(final String serverTemplateId, fi
     }
 
     @Override
-    public String startCaseInstance(final String serverTemplateId, final String containerId, final String caseDefinitionId, final String owner) {
-        final CaseFile caseFile = CaseFile.builder().addUserAssignments(CASE_OWNER_ROLE, owner).build();
-        return client.startCase(containerId, caseDefinitionId, caseFile);
+    public String startCaseInstance(final String serverTemplateId, final String containerId, final String caseDefinitionId, final String owner,
+                                    final List<CaseRoleAssignmentSummary> roleAssignments) {
+        final CaseFile.Builder builder = CaseFile.builder();
+        builder.addUserAssignments(CASE_OWNER_ROLE, owner);
+        roleAssignments.stream().forEach( a -> {
+            a.getGroups().stream().forEach(g -> builder.addGroupAssignments(a.getName(), g));
+            a.getUsers().stream().forEach(u -> builder.addUserAssignments(a.getName(), u));
+        });
```

Comment:
FYI Unless you actually need the streams,`Collection.forEach()` will do the same thing.